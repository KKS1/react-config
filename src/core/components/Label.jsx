import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
