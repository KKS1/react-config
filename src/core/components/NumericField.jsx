import React, { Component } from "react";
import MaskedInput from "./MaskedInput";
import Label from "./Label";

export default class NumericField extends Component {
  static defaultProps = {
    label: "Label",
    placeholder: "",
    name: "numericFormField",
    onChange: () => {}
  };

  changeHandler = e => {
    this.props.onChange(e.target.value);
  };

  render() {
    const {
      label,
      placeholder,
      name,
      value,
      id,
      options,
      disabled
    } = this.props;
    const { maskconfig } = options || {};

    return (
      <div className="c-label-field">
        <Label {...this.props} />
        <MaskedInput
          id={id}
          className="c-label-field__input"
          type="text"
          inputMode="numeric"
          placeholder={placeholder}
          name={name}
          disabled={disabled}
          value={value == null ? 0 : value}
          maskconfig={maskconfig}
          onChange={this.changeHandler}
        />
      </div>
    );
  }
}
