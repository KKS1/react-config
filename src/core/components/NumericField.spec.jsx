import React, { Component } from "react";
import Label from "./Label";
import NumericField from "./NumericField";

describe("NumericField", () => {
  test("render_NumericField", () => {
    const wrapper = shallow(<NumericField />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("attach_props", () => {
    const props = {
      id: "formOne_labelOne",
      value: 100,
      options: {
        maskconfig: {
          min: 0,
          max: 100000,
          digits: 2
        }
      }
    };
    const wrapper = shallow(<NumericField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();
  });

  test("call_onChange_prop", () => {
    const changeHandler = jest.fn();

    const props = {
      id: "formOne_labelOne",
      value: 100,
      onChange: changeHandler
    };
    const wrapper = shallow(<NumericField {...props} />);
    const instance = wrapper.instance();
    expect(instance).toMatchSnapshot();

    instance.changeHandler({ target: { value: 200 } });

    expect(changeHandler).toHaveBeenCalledWith(200);
  });
});
