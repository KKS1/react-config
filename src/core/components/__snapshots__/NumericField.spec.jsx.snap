// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericField attach_props 1`] = `
NumericField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "numericFormField",
    "onChange": [Function],
    "options": Object {
      "maskconfig": Object {
        "digits": 2,
        "max": 100000,
        "min": 0,
      },
    },
    "placeholder": "",
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <NumericField
        id="formOne_labelOne"
        label="Label"
        name="numericFormField"
        onChange={[Function]}
        options={
          Object {
            "maskconfig": Object {
              "digits": 2,
              "max": 100000,
              "min": 0,
            },
          }
        }
        placeholder=""
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="numericFormField"
          onChange={[Function]}
          options={
            Object {
              "maskconfig": Object {
                "digits": 2,
                "max": 100000,
                "min": 0,
              },
            }
          }
          placeholder=""
          value={100}
        />
        <MaskedInput
          className="c-label-field__input"
          id="formOne_labelOne"
          inputMode="numeric"
          maskconfig={
            Object {
              "digits": 2,
              "max": 100000,
              "min": 0,
            }
          }
          name="numericFormField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value={100}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`NumericField call_onChange_prop 1`] = `
NumericField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "id": "formOne_labelOne",
    "label": "Label",
    "name": "numericFormField",
    "onChange": [MockFunction],
    "placeholder": "",
    "value": 100,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <NumericField
        id="formOne_labelOne"
        label="Label"
        name="numericFormField"
        onChange={[MockFunction]}
        placeholder=""
        value={100}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          id="formOne_labelOne"
          label="Label"
          name="numericFormField"
          onChange={[MockFunction]}
          placeholder=""
          value={100}
        />
        <MaskedInput
          className="c-label-field__input"
          id="formOne_labelOne"
          inputMode="numeric"
          maskconfig={
            Object {
              "alias": "numeric",
              "autoGroup": true,
              "digits": 0,
              "digitsOptional": true,
              "groupSeparator": ",",
              "min": 0,
              "rightAlign": false,
            }
          }
          name="numericFormField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value={100}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`NumericField render_NumericField 1`] = `
NumericField {
  "changeHandler": [Function],
  "context": Object {},
  "props": Object {
    "label": "Label",
    "name": "numericFormField",
    "onChange": [Function],
    "placeholder": "",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <NumericField
        label="Label"
        name="numericFormField"
        onChange={[Function]}
        placeholder=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="c-label-field"
      >
        <Label
          label="Label"
          name="numericFormField"
          onChange={[Function]}
          placeholder=""
        />
        <MaskedInput
          className="c-label-field__input"
          inputMode="numeric"
          maskconfig={
            Object {
              "alias": "numeric",
              "autoGroup": true,
              "digits": 0,
              "digitsOptional": true,
              "groupSeparator": ",",
              "min": 0,
              "rightAlign": false,
            }
          }
          name="numericFormField"
          onChange={[Function]}
          placeholder=""
          type="text"
          value={0}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
